packages/xstate-fsm/src/index.ts

```diff
@@ -170,22 +170,31 @@ export function createMachine<
       }
 
       if (stateConfig.on) {
-        const transitions = toArray(stateConfig.on[eventObject.type]);
+        const transitions: Array<
+          StateMachine.Transition<TContext, TEvent>
+        > = toArray(stateConfig.on[eventObject.type]);
 
         for (const transition of transitions) {
           if (transition === undefined) {
             return createUnchangedState(value, context);
           }
 
-          const { target = value, actions = [], cond = () => true } =
+          const { target, actions = [], cond = () => true } =
             typeof transition === 'string'
               ? { target: transition }
               : transition;
 
+          const isInternal = target === undefined;
```

Comment:
Those shouldn't be confused with internal transitions - internal transitions in SCXML and XState can have targets

```suggestion
          const isTargetless = target === undefined;
```